packages/streaming-image-volume-loader/src/StreamingImageVolume.ts

```diff
@@ -83,18 +92,49 @@ export default class StreamingImageVolume extends ImageVolume {
     };
   }
 
-  private _hasLoaded = (): boolean => {
-    const { loadStatus, imageIds } = this;
-    const numFrames = imageIds.length;
+  protected initialize(): void {
+    this._createCornerstoneImageMetaData();
+  }
+
+  // The frameIndex is equal to imageIdIndex for 3D datasets
+  // but they are different for 4D datasets
+  protected imageIdIndexToFrameIndex(imageIdIndex: number): number {
+    return imageIdIndex % this.numFrames;
+  }
+
+  public get numFrames(): number {
+    return this.imageIds.length;
+  }
+
+  public getScalarDataArrays(): Array<Float32Array | Uint8Array> {
+    return [this.scalarData];
+  }
+
+  public getActiveScalarData(): Float32Array | Uint8Array {
+    return this.scalarData;
+  }
+
+  public getScalarDataByImageIdIndex(
+    imageIdIndex: number
+  ): Float32Array | Uint8Array {
+    if (imageIdIndex < 0 || imageIdIndex >= this.imageIds.length) {
+      throw new Error('imageIdIndex out of range');
+    }
+
+    return this.scalarData;
+  }
+
+  public invalidateVolume(): void {
+    const { imageData, vtkOpenGLTexture } = this;
+    const { numFrames } = this;
 
     for (let i = 0; i < numFrames; i++) {
-      if (!loadStatus.cachedFrames[i]) {
-        return false;
-      }
+      vtkOpenGLTexture.setUpdatedFrame(i);
     }
 
-    return true;
-  };
+    imageData.modified();
+    autoLoad(this.volumeId);
+  }
```

Comment:
this should accept a param as `immediate` or something similar to decide whether to call `autoLoad` or not